kube-client/src/config/mod.rs

```diff
@@ -271,6 +273,45 @@ impl Config {
         })
     }
 
+    /// Override configuration based on environment variables
+    ///
+    /// This is only intended for use as a debugging aid, and the specific variables and their behaviour
+    /// should **not** be considered stable across releases.
+    ///
+    /// Currently, the following overrides are supported:
+    ///
+    /// - `KUBE_RS_DEBUG_IMPERSONATE_USER`: A Kubernetes user to impersonate, for example: `system:serviceaccount:default:foo` will impersonate the `ServiceAccount` `foo` in the `Namespace` `default`
+    /// - `KUBE_RS_DEBUG_IMPERSONATE_GROUP`: A Kubernetes group to impersonate, multiple groups may be specified by separating them with commas
+    /// - `KUBE_RS_DEBUG_OVERRIDE_URL`: A Kubernetes cluster URL to use rather than the one specified in the config, useful for proxying traffic through `kubectl proxy`
+    #[tracing::instrument(level = "warn")]
+    pub fn apply_debug_overrides(&mut self) {
+        // Log these overrides loudly, to emphasize that this is only a debugging aid, and should not be relied upon in production
```

Comment:
if we word it like this then we should not be putting it inside the main `Config::infer`

Reply:
My main inspiration here has been `SSLKEYLOGFILE`, which all prod Firefox and Chrome builds listen for and respect, but which no "normal" user would use.

As long as the user never sets any of the debug envvars then this should be completely inert.

Reply:
yeah, i think the behaviour here is fine - it was mainly the naming that concerned me a bit.
but re-reading it now it is a lot more neutral than i interpreted it this morning :thinking: 

i'll resolve these.